src/main/java/core/basesyntax/FileWork.java

```
@@ -8,7 +16,29 @@
  * Результат: web wide width world</p>
  */
 public class FileWork {
-    public String[] readFromFile(String fileName) {
-        return null;
+
+    private static final String REGEX_W = "\\bw\\w*\\b";
+
+    public String[] readFromFile(String fileName) throws RuntimeException {
+
+        ArrayList<String> listOfWords = new ArrayList<>();
+
+        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                Pattern patternRegexW = Pattern.compile(REGEX_W);
+                Matcher lineToLowerCase = patternRegexW.matcher(line.toLowerCase());
+                while (lineToLowerCase.find()) {
+                    listOfWords.add(lineToLowerCase.group().toLowerCase());
+                }
+            }
+            Collections.sort(listOfWords);
+        } catch (IOException exception) {
+            exception.printStackTrace();
```

let's rethrow this Exception as a RuntimeException